source/extensions/omni.isaac.lab/omni/isaac/lab/envs/mdp/observations.py

```diff
@@ -279,65 +283,89 @@ def image(
+        self.model_name: str = cfg.params.get("model_name", "ResNet18")  # type: ignore
+        self.model_device: str = cfg.params.get("model_device", env.device)  # type: ignore
+
+        # List of Theia models - These are configured through `_prepare_theia_transformer_model` function
+        default_theia_models = [
             "theia-tiny-patch16-224-cddsv",
             "theia-tiny-patch16-224-cdiv",
             "theia-small-patch16-224-cdiv",
             "theia-base-patch16-224-cdiv",
             "theia-small-patch16-224-cddsv",
             "theia-base-patch16-224-cddsv",
         ]
-
-        # List of ResNet models
-        resnet_models = ["resnet18", "resnet34", "resnet50", "resnet101"]
-
-        self.default_model_zoo_cfg = {}
-
-        # Add Theia models to the zoo
-        for model_name in theia_models:
-            self.default_model_zoo_cfg[model_name] = create_theia_model(model_name)
-
-        # Add ResNet models to the zoo
-        for resnet_name in resnet_models:
-            self.default_model_zoo_cfg[resnet_name] = create_resnet_model(resnet_name)
-
-        self.model_zoo_cfg = self.default_model_zoo_cfg
-        self.model_zoo = {}
+        # List of ResNet models - These are configured through `_prepare_resnet_model` function
+        default_resnet_models = ["resnet18", "resnet34", "resnet50", "resnet101"]
+
+        # Check if model name is specified in the model zoo configuration
+        if self.model_zoo_cfg is not None and self.model_name not in self.model_zoo_cfg:
+            raise ValueError(
+                f"Model name '{self.model_name}' not found in the provided model zoo configuration."
+                " Please add the model to the model zoo configuration or use a different model name."
+                f" Available models in the provided list: {list(self.model_zoo_cfg.keys())}."
+                "\nHint: If you want to use a default model, consider using one of the following models:"
+                f" {default_theia_models + default_resnet_models}. In this case, you can remove the"
+                " 'model_zoo_cfg' parameter from the observation term configuration."
+            )
+        if self.model_zoo_cfg is None:
+            if self.model_name in default_theia_models:
+                model_config = self._prepare_theia_transformer_model(self.model_name, self.model_device)
+            elif self.model_name in default_resnet_models:
+                model_config = self._prepare_resnet_model(self.model_name, self.model_device)
+            else:
+                raise ValueError(
+                    f"Model name '{self.model_name}' not found in the default model zoo configuration."
+                    f" Available models: {default_theia_models + default_resnet_models}."
+                )
+        else:
+            model_config = self.model_zoo_cfg[self.model_name]
+
+        # Retrieve the model, preprocess and inference functions
+        self._model = model_config["model"]()
```

Comment:
By keeping pre-process separate, I think this leaves room for using hydra to override the pre-process to play around with different pre-processing while keeping the model the same. For example we could provide an idempotent pre-process that only divides by 255 instead of normalizing to training set for experimentation

Reply:
kwargs could work as per convo with @Mayankm96

Reply:
Done